### ISSUE TITLE
Metrics of http client may have memory/size problem

### ISSUE THREAD

**Issue by kojilin** [line/armeria#278]
https://github.com/line/armeria/pull/261 has some problem if client access path is like
http://localhost/parameter?=${variable}  or http://localhost/parameter/${variable}
It's not suitable when variable has lots variation.

**Comment by trustin** [comment]
I think both server and client are affected by this issue.

- For the server-side, we could use some string representation of `PathMapping` of a `Service` instead of a path. We may want to make it configurable.
- For the client-side, I'm not sure. Maybe we should make it configurable?

**Comment by kojilin** [comment]
Yes, I think maybe both client/server need to be configurable if it's not RPC.
ex:
```
(path, method)->{
  if(path.matches("/foo/.*/bar"){
    return "/foo/${id}/var";
  }
 return null; //null means don't log.
}
```

or just provide patterns array and log when matched.

**Comment by trustin** [comment]
Will fix this today. Stay tuned for a PR.

**Comment by trustin** [comment]
Fixed via #280
